pkg/streamingpromql/planning.go

```diff
@@ -440,6 +477,51 @@ func findFunction(name string) (functions.Function, bool) {
 	return functions.FUNCTION_UNKNOWN, false
 }
 
+// functionNeedsDeduplication checks if a function needs deduplication and merge operator.
+// This is determined by whether the function drops the __name__ label or otherwise manupulates it.
+func functionNeedsDeduplication(fnc functions.Function) bool {
```

Comment:
It'd be good to have some way to ensure we don't forget to add a newly-added function here.

Perhaps this `switch` should be expanded to cover all `functions.Function`s and should panic if `fnc` is not listed? Then we could have a unit test that checks all known functions don't cause `functionNeedsDeduplication` to panic. 

Or maybe use a map?